composer/airflow_1_samples/gke_operator.py

```diff
@@ -0,0 +1,243 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# [START composer_gkeoperator_airflow_1]
+
+
+from airflow import models
+from airflow.operators.bash_operator import BashOperator
+from airflow.providers.google.cloud.operators.kubernetes_engine import (
+    GKECreateClusterOperator,
+    GKEDeleteClusterOperator,
+    GKEStartPodOperator,
+)
+from airflow.utils.dates import days_ago
+
+
+with models.DAG(
+    "example_gcp_gke",
+    schedule_interval=None,  # Override to match your needs
+    start_date=days_ago(1),
+    tags=["example"],
+) as dag:
+
+    # [START composer_gke_create_cluster_airflow_1]
+    # [START composer_gkeoperator_minconfig_airflow_1]
+    # [START composer_gkeoperator_templateconfig_airflow_1]
+    # [START composer_gkeoperator_affinity_airflow_1]
+    # [START composer_gkeoperator_fullconfig_airflow_1]
+    # TODO(developer): update with your values
+    PROJECT_ID = "my-project-id"
+    CLUSTER_ZONE = "us-west1-a"
+    CLUSTER_NAME = "example-cluster"
+    # [END composer_gkeoperator_minconfig_airflow_1]
+    # [END composer_gkeoperator_templateconfig_airflow_1]
+    # [END composer_gkeoperator_affinity_airflow_1]
+    # [END composer_gkeoperator_fullconfig_airflow_1]
+    CLUSTER = {"name": CLUSTER_NAME, "initial_node_count": 1}
+    # [END composer_gke_create_cluster_airflow_1]
+    # [START composer_gke_create_cluster_airflow_1]
+    create_cluster = GKECreateClusterOperator(
+        task_id="create_cluster",
+        project_id=PROJECT_ID,
+        location=CLUSTER_ZONE,
+        body=CLUSTER,
+    )
+
```

Comment:
```suggestion
# Using the BashOperator to create NodePools is a workaround. In Airflow 2, because of https://github.com/apache/airflow/pull/17820 this can be done as part of the operator configuration
```